### PR TITLE
chore: update a few names

### LINKED ISSUE #183: Rename `SimpleCacheClient` to `CacheClient`

### LINKED ISSUE #184: Rename `Dictionary` `Items` → `Elements`

### LINKED ISSUE #185: Use `Ttl` and `CollectionTtl` consistently.
* We should also make sure Ttl is consistently named through out SDK

### PR DESCRIPTION
**Pull request by poppoerika**
Rename the following:
- `SimpleCacheClient` to `CacheClient`
- dictionary `items` to `elements`
- `CollectionTTL` to `CollectionTtl` 

Closes #183 
Closes #184 
Closes #185

### CODE REVIEW

File: momento/dictionary_test.go [hunk 1]
```diff
@@ -5,9 +5,6 @@ import (
 	"time"
 
 	"github.com/google/uuid"
-	. "github.com/onsi/ginkgo/v2"
-	. "github.com/onsi/gomega"
-
```
Comment (schwern) [inline momento/dictionary_test.go:10]: These are where Describe and It and Expect come from.

File: momento/dictionary_get_fields.go [hunk 2]
```diff
@@ -26,7 +26,7 @@ func (resp DictionaryGetFieldsHit) ValueMap() map[string]string {
 
 func (resp DictionaryGetFieldsHit) ValueMapStringString() map[string]string {
 	ret := make(map[string]string)
-	for idx, item := range resp.items {
+	for idx, item := range resp.elements {
```
Comment (schwern) [inline momento/dictionary_get_fields.go:29]:
```suggestion
	for idx, element := range resp.elements {
```

File: momento/dictionary_get_fields.go [hunk 3]
```diff
@@ -36,7 +36,7 @@ func (resp DictionaryGetFieldsHit) ValueMapStringString() map[string]string {
 
 func (resp DictionaryGetFieldsHit) ValueMapStringBytes() map[string][]byte {
 	ret := make(map[string][]byte)
-	for idx, item := range resp.items {
+	for idx, item := range resp.elements {
```
Comment (schwern) [inline momento/dictionary_get_fields.go:39]:
```suggestion
	for idx, element := range resp.elements {
```

File: momento/dictionary_set_fields.go [hunk 4]
```diff
@@ -47,13 +47,13 @@ func (r *DictionarySetFieldsRequest) initGrpcRequest(client scsDataClient) error
...
 	}
 
-	var items map[string][]byte
-	if items, err = prepareItems(r); err != nil {
+	var elements map[string][]byte
+	if elements, err = prepareItems(r); err != nil {
 		return err
 	}
 
 	var pbItems []*pb.XDictionaryFieldValuePair
```
Comment (schwern) [inline momento/dictionary_set_fields.go:55]: pbElements?

File: momento/simple_cache_client.go [hunk 5]
```diff
@@ -74,8 +74,8 @@ type SimpleCacheClientProps struct {
 	DefaultTTL         time.Duration
 }
 
-// NewSimpleCacheClient returns a new SimpleCacheClient with provided authToken, DefaultTTLSeconds, and opts arguments.
-func NewSimpleCacheClient(props *SimpleCacheClientProps) (SimpleCacheClient, error) {
+// NewSimpleCacheClient returns a new CacheClient with provided authToken, DefaultTTLSeconds, and opts arguments.
+func NewSimpleCacheClient(props *SimpleCacheClientProps) (CacheClient, error) {
```
Comment (schwern) [inline momento/simple_cache_client.go:78]: NewCacheClient and CacheClientProps.

**Review by schwern — CHANGES_REQUESTED** [review]

File: momento/dictionary_fetch.go [hunk 6]
```diff
@@ -13,7 +13,7 @@ type DictionaryFetchResponse interface {
 }
 
 type DictionaryFetchHit struct {
-	items             map[string][]byte
+	elements          map[string][]byte
 	itemsStringString map[string]string
```
[the remark this answers is not in the document]
Comment (eaddingtonwhite) [reply momento/dictionary_fetch.go:17]: This seem bit redundant to me. Isnt code a bit self documenting here? `elementsStringString map[string]string`

File: momento/dictionary_test.go (hunk 1 above)
Comment (poppoerika) [reply momento/dictionary_test.go:10]: Wow, how did I remove that. Thank you.

File: momento/dictionary_fetch.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (poppoerika) [reply momento/dictionary_fetch.go:17]: We have in `_dictionaryStringString` in our .NET SDK ([link](https://github.com/momentohq/client-sdk-dotnet/blob/main/src/Momento.Sdk/Responses/CacheDictionaryGetFieldsResponse.cs#L121)).
So I think we want to follow the pattern here as well? I believe the decision of  including type names in variable names came from the bug bash.

File: momento/dictionary_set_fields.go (hunk 4 above)
Comment (poppoerika) [reply momento/dictionary_set_fields.go:55]: Also `prepareElements` 😮‍💨

File: momento/dictionary_fetch.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (eaddingtonwhite) [reply momento/dictionary_fetch.go]: Ok I agree let's keep consistent then. It's internal variable so can always iterate on it in future if we change our minds.

**Review by schwern — CHANGES_REQUESTED** [review]
Looks like a few SimpleCacheClients were missed. Have you seen `make precommit`? It does all the linting and tests.

**Pull request merged**